frappe/database/__init__.py

```diff
@@ -71,7 +75,11 @@ def get_command(host=None, port=None, user=None, password=None, db_name=None, ex
 		else:
 			bin, bin_name = which("psql"), "psql"
 
-		if password:
+		if socket and password:
+			conn_string = f"postgresql://{user}:{password}@/{db_name}?host={socket}"
+		elif socket:
+			conn_string = f"postgresql://{user}@/{db_name}?host={socket}"
```

Comment:
That behavior seems according to spec, for postgres: https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING:~:text=the%20value%20is%20the%20name%20of%20the%20directory%20in%20which%20the%20socket%20file%20is%20stored

Reply:
Fortunately, `psycopg2` is consistent: https://www.psycopg.org/docs/extensions.html#psycopg2.extensions.ConnectionInfo.host:~:text=or%20a%20directory%20path%20if%20the%20connection%20is%20via%20Unix%20socket

Reply:
This is how I addressed it: 83b9da06e2 &mdash; I think that's also the most sensible thing we can do, short of creating a private api.

Reply:
Will this week be the week that we'll be freed from carrying this PR? :smile: :innocent:

Reply:
I have another diff, but this would be a gain in function. I suppose it could be a remedy of an oversight, but it might as well be the way it is for genuine reasons, so I'd prefer to present it in a follow up PR for discussion. It's a small one, though.

Regardless, in case it's the way it is for genuine reasons that discussion would be inappropriate for this PR here.

Reply:
> Will this week be the week that we'll be freed from carrying this PR? 😄 😇

Will try to merge within the next 1-2 weeks, I discovered an issue with postgres while testing this, once I get time to fix that will resume testing this.

Reply:
Maybe I can help out in any way?

Reply:
Basically [this](https://github.com/frappe/frappe/blob/develop/frappe/database/postgres/setup_db.py#L24) regex seemed to be wrong, realised because I failed to setup a new site with a fresh postgres instance.

Simple fix, will do it sometime soon and go through this again.

Reply:
@akhilnarang I've seen that some of the key refacorings leading up to this PR  have recently made it into v15 due to a performance (?) fix that you did which dependent on them. I've also just migrated to v15 and thus our production environment starts converging. I think this would be a good moment to get a final version of this PR locked and merged so that I can then continue testing it thoroughly in all possible scenarios still for the rest of the v15 release cycle.

Reply:
Yep basically all the perf related changes were backrported (primarily related to not extracting the entire backup file). There were a bunch of small dependent fixes - so I grabbed those in as well.

Will try to get back to this ASAP.

Reply:
@akhilnarang Do you think you could kindly have a look at this again, maybe this week?